### PR TITLE
[WIP] Add TS support to react-native

### PR DESCRIPTION
**Pull request by fabioh8010**
TODO

### CODE REVIEW

File: packages/react-native-web/package.json [hunk 1]
```diff
@@ -49,5 +51,8 @@
     "react-component",
     "react-native",
     "web"
-  ]
+  ],
+  "devDependencies": {
+    "react-native": "^0.74.3"
+  }
```
Comment (blazejkustra) [inline packages/react-native-web/package.json:57]: I don't think we need to add this line as part of this PR

File: packages/react-native-web/package.json [hunk 2]
```diff
@@ -19,6 +20,7 @@
     "build:commonjs": "cross-env BABEL_ENV=commonjs babel src --config-file ../../configs/babel.config.js --out-dir dist/cjs --ignore \"**/__tests__\"",
     "build:es": "babel src --config-file ../../configs/babel.config.js --out-dir dist --ignore \"**/__tests__\"",
     "build:flow": "gen-flow-files ./src --out-dir ./dist",
+    "build:typescript": "tsc",
```
Comment (blazejkustra) [inline packages/react-native-web/package.json:23]: Let's remove this step and tsconfig file, I think it's not needed 👍

**Review by blazejkustra — CHANGES_REQUESTED** [review]

**Comment by blazejkustra** [comment]
Awesome work @fabioh8010! Let's update the description to address why the PR is needed in the first place, and post it [here](https://github.com/necolas/react-native-web/issues/832) and [here](https://github.com/necolas/react-native-web/issues/1684) to gather some 👀

**Review by blazejkustra — APPROVED** [review]

**Comment by necolas** [comment]
I'm not going to merge and maintain a separate set of TS types. Sorry

**Comment by blazejkustra** [comment]
> I'm not going to merge and maintain a separate set of TS types. Sorry

That is fine 😄 This PR can be closed now, as we just added TS definitions for RNW and extended RN types in [DefinietlyTyped repo](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70147)!

**Comment by blazejkustra** [comment]
This PR can be closed now, as we just added TS definitions for RNW and extended RN types in [DefinietlyTyped repo](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70147). 

### Installation
Simply install the types with `npm run install @types/react-native-web` and supply `react-native-web` as a member of the types compiler option:
```ts
{
  "compilerOptions": {
    "types": ["react-native-web"]
  }
}
```

**Comment by fabioh8010** [comment]
Closing as per discussed above, we will create a PR with the instructions to this repo soon!

**Pull request closed without merging**